src/azul/collections.py

```diff
@@ -271,3 +273,64 @@ def to_dict(self):
             k: v.to_dict() if isinstance(v, NestedDict) else v
             for k, v in self.items()
         }
+
+
+class OrderedSet(MutableSet[Any]):
```

Comment:
Use a type variable instead of `Any`. Probably best to reuse `K` or `V` from earlier in the module. I'm not sure which is better in this context.

```suggestion
class OrderedSet(MutableSet[K]):
```

And replace occurrences of `Any` throughout the class with the same variable.